oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -107,8 +109,14 @@ pub fn report_event<T: AsRef<str>>(label: T) -> Result<(), OakStatus> {
 /// See [`report_metric`](https://github.com/project-oak/oak/blob/main/docs/oak_functions_abi.md#report_metric).
 #[cfg(feature = "oak-metrics")]
 pub fn report_metric<T: AsRef<str>>(label: T, value: i64) -> Result<(), OakStatus> {
-    let buf = label.as_ref().as_bytes();
-    let status = unsafe { oak_functions_abi::report_metric(buf.as_ptr(), buf.len(), value) };
+    let label = String::from(label.as_ref());
```

Comment:
I think this is equivalent, but for consistency we usually use `to_owned` to convert `&str` to `String`, e.g. `label.as_ref().to_owned()`.